test/UtilsTest.js

```diff
@@ -244,212 +245,124 @@ describe('FlowLimit', async () => {
+        const addr = test.address;
 
-    it('Should revert when flow drift exceeds limit after in/out/in/in', async () => {
-        await test.setFlowLimit(10).then((tx) => tx.wait);
-        await test.addFlowIn(10).then((tx) => tx.wait);
-        await test.addFlowOut(10).then((tx) => tx.wait);
-        await test.addFlowIn(1).then((tx) => tx.wait);
-
-        await expectRevert((gasOptions) => test.addFlowIn(10, gasOptions), test, 'FlowLimitExceeded', [10, 11, test.address]);
-    });
-
-    it('Should revert when flow slightly exceeds large limit', async () => {
-        const limit = MaxUint256.sub(1);
-
-        await test.setFlowLimit(limit).then((tx) => tx.wait);
-        await test.addFlowIn(limit).then((tx) => tx.wait);
-
-        await expectRevert((gasOptions) => test.addFlowIn(1, gasOptions), test, 'FlowLimitExceeded', [limit, limit.add(1), test.address]);
-    });
-
-    it('Should revert when out-flow drift exceeds max limit', async () => {
-        const limit = MaxUint256.sub(2);
-
-        await test.setFlowLimit(limit).then((tx) => tx.wait);
-        await test.addFlowIn(1).then((tx) => tx.wait);
-        await test.addFlowOut(limit).then((tx) => tx.wait);
-
-        await expectRevert((gasOptions) => test.addFlowOut(2, gasOptions), test, 'FlowLimitExceeded', [limit, limit.add(1), test.address]);
-    });
-
-    it('Should revert when in/out/in combination creates net imbalance over the limit', async () => {
-        const half = MaxUint256.div(2);
-        const limit = half;
+        if (error === 'FlowAmountOverflow') {
+            const current = isIn ? await test.flowInAmount() : await test.flowOutAmount();
+            return [amount, BigNumber.from(current), addr];
+        } else if (error === 'FlowLimitExceeded') {
+            const forwardFlow = isIn ? flowIn.add(amount) : flowOut.add(amount);
+            const reverseFlow = isIn ? flowOut : flowIn;
+            const netFlow = forwardFlow.gt(reverseFlow) ? forwardFlow.sub(reverseFlow) : reverseFlow.sub(forwardFlow);
+            return [flowLimit, netFlow, addr];
+        }
 
-        await test.setFlowLimit(limit).then((tx) => tx.wait);
-        await test.addFlowIn(half).then((tx) => tx.wait);
-        await test.addFlowOut(1).then((tx) => tx.wait);
+        // Default case: FlowAmountExceededLimit
+        return [flowLimit, amount, addr];
+    };
+
+    const executeTestCases = (error, testCases) => {
+        const allCases = [
+            ...testCases,
+            ...testCases.map(({ flowLimit, flows }) => ({
+                flowLimit,
+                flows: flipFlows(flows),
+            })),
+        ];
 
-        await expectRevert((gasOptions) => test.addFlowIn(2, gasOptions), test, 'FlowLimitExceeded', [limit, half.add(1), test.address]);
-    });
+        for (const { flowLimit, flows } of allCases) {
+            const desc = `Should revert with ${error} (flowLimit=${flowLimit}, steps=${flows.length})`;
+
+            it(desc, async () => {
```

Comment:
```suggestion
            it(`Should revert with ${error} (flowLimit=${flowLimit}, steps=${flows.length})`, async () => {
```
intermediate var not needed?